src/TimePicker/Clock.js

```diff
@@ -20,10 +20,16 @@ class Clock extends Component {
   };
 
   state = {
-    selectedTime: this.props.initialTime || new Date(),
+    selectedTime: null,
     mode: 'hour',
   };
 
+  componentWillMount() {
+    this.setState({
+      selectedTime: this.props.initialTime,
```

Comment:
` || new Date(),` is missing, the behavior is not identical. E.g. `this.props.initialTime = null`